File: dart/example/basic_extraction.dart
// basic_extraction — build a PDF from Markdown, then extract it back.
// Run in CI as a smoke example (no external fixture).
import 'package:pdf_oxide/pdf_oxide.dart';

void main() {
  final pdf = Pdf.fromMarkdown(
      '# Hello pdf_oxide\n\nThis is a **Dart** binding smoke example.\n');
  final doc = PdfDocument.openFromBytes(pdf.toBytes());
  try {
    print('pages:   ${doc.pageCount}');
    print('version: ${doc.version}');
    print('--- text (page 0) ---');
    print(doc.extractText(0));
    print('--- markdown (all) ---');
    print(doc.toMarkdownAll());
  } finally {
    doc.close();
    pdf.close();
  }
}

File: r/inst/examples/basic_extraction.R
# basic_extraction — build a PDF from Markdown, then extract it back.
# Run in CI as a smoke example (no external fixture).
library(pdfoxide)

pdf <- pdf_from_markdown(
  "# Hello pdf_oxide\n\nThis is an **R** binding smoke example.\n")
doc <- pdf_open_from_bytes(pdf_to_bytes(pdf))

cat("pages:  ", pdf_page_count(doc), "\n")
v <- pdf_version(doc)
cat("version:", paste(v$major, v$minor, sep = "."), "\n")
cat("--- text (page 0) ---\n", pdf_extract_text(doc, 0), "\n", sep = "")
cat("--- markdown (all) ---\n", pdf_to_markdown_all(doc), "\n", sep = "")
